### DISCUSSION TITLE
TensorRT converted yolov5 has multiple outputs

### DISCUSSION THREAD

**Discussion by bdv29** [ultralytics/yolov5#3922] [Q&A]
So I converted a custom trained model to TensorRT to run on the Jetson nano, 
and the output I get is a list with four arrays, 
the fourth array seems to be the one that gives me the correct bounding boxes when reshaped correctly taking account the amount of classes  I have trained it for and fed into the NMS function , 
but I'm wondering is what are the other three arrays for? Is it just some odd idiosyncrasy of tensorRT or is there another aspect of yolo that I don't understand and could those other arrays have useful information?

screenshot of the four arrays 
![image](https://user-images.githubusercontent.com/87079397/124773026-2e9e7f80-df02-11eb-995a-cf030e708057.png)

**Comment by glenn-jocher** [comment]
@bdv29 0, 1 and 2 in your screenshot are part of 3. See **TensorRT Deployment** Tutorial for exporting to TensorRT:

## YOLOv5 Tutorials

* [Train Custom Data](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://github.com/ultralytics/yolov5/wiki/Tips-for-Best-Training-Results)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://github.com/ultralytics/yolov5/issues/2518)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://github.com/ultralytics/yolov5/issues/475)
* [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36)&nbsp; ⭐ NEW
* [TorchScript, ONNX, CoreML Export](https://github.com/ultralytics/yolov5/issues/251) 🚀
* [Test-Time Augmentation (TTA)](https://github.com/ultralytics/yolov5/issues/303)
* [Model Ensembling](https://github.com/ultralytics/yolov5/issues/318)
* [Model Pruning/Sparsity](https://github.com/ultralytics/yolov5/issues/304)
* [Hyperparameter Evolution](https://github.com/ultralytics/yolov5/issues/607)
* [Transfer Learning with Frozen Layers](https://github.com/ultralytics/yolov5/issues/1314)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)
